doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,241 @@
+     weak learners ([`DecisionTreeRegressor`s](decision_tree_regressor.md)).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical and/or categorical data, using the given hyperparameters.
+
+---
+
+#### Constructor Parameters:
+
+| **name** | **type** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `data` | [`arma::mat`](../matrices.md) | [Column-major](../matrices.md#representing-data-in-mlpack) training matrix. | _(N/A)_ |
+| `datasetInfo` | [`data::DatasetInfo`](https://github.com/mlpack/mlpack/blob/master/doc/user/load_save.md#loading-categorical-data) | Dataset information, specifying type information for each dimension. | _(N/A)_ |
+| `labels` | [`arma::Row<size_t>`](../matrices.md) | Training labels, [between `0` and `numClasses - 1`](../load_save.md#normalizing-labels) (inclusive).  Should have length `data.n_cols`.  | _(N/A)_ |
+| `numClasses` | `size_t` | Number of classes in the dataset. | _(N/A)_ |
+| `numWeakLearners` | `size_t` | Number of weak learner models to be used. | 5 |
+| `minimumLeafSize` | `size_t` | Minimum leaf size for weak learner decision tree. | `10` |
+| `minimumGainSplit` | `double` | Minimum gain split for weak learner decision tree | `1e-7` |
+| `maximumDepth` | `size_t` | Maximum tree depth for weak learner decision tree | `2` |
+
+
+### Training
+
+If training is not done as part of the constructor call, it can be done with one
+of the following versions of the `Train()` member function:
+
+ * `gb.Train(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical data without using any decision tree arguments.
+
+---
+
+ * `gb.Train(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on mixed categorical data using the `minimumLeafSize`, `minimumGainSplit`, and `maximumDepth` options to control the behavior of the weak learners ([`DecisionTreeRegressor`s](decision_tree_regressor.md)).
+
+---
+
+Types of each argument are the same as in the table for constructors
+[above](#constructor-parameters).
+
+***Note:***: Training is not incremental.  A second call to `Train()` will retrain the
+model from scratch.
+
+### Classification
+
+Once a `GradBoosting` object is trained, the `Classify()` member function can be used
+to make class predictions for new data.
+
+ * `size_t predictedClass = gb.Classify(point)`
+    - ***(Single-point)***
+    - Classify a single point, returning the predicted class.
+
+---
+
+ * `gb.Classify(point, prediction)`
+    - ***(Single-point)***
+    - Classify a single point and compute class probabilities.
+    - The predicted class is stored in `prediction`.
+
+---
+
+ * `gb.Classify(data, predictedLabels)`
+    - ***(Multi-point)***
+    - Classify a set of points.
+    - The prediction for data point `i` can be accessed with `predictedLabels[i]`.
```

Comment:
What about a multi-point overload that computes class probabilities too?  I see it is not implemented, but I imagine it would be easy to implement, and it would be good to have because it would make this class consistent with mlpack's other classifiers.